src/helpers.c

```diff
@@ -622,17 +627,44 @@ int strtoi_err(const char *str, char **endptr, int *value)
 	 * and return an overflow error code. */
 	if (num > INT_MAX) {
 		num = INT_MAX;
-		err = -EOVERFLOW;
+		err = -ERANGE;
 	} else if (num < INT_MIN) {
 		num = INT_MIN;
-		err = -EOVERFLOW;
+		err = -ERANGE;
 	}
 
 	*value = (int)num;
 
 	return err;
 }
 
+/* strtoi_err: Safely convert and string to integer avoiding overflows
+ *
+ * The strtol function is commonly used to convert a string to a number and
+ * the result is frequently stored in an int type, but type casting a long to
+ * an int can cause overflows.
+ *
+ * This function returns negative error codes based on the errno table:
+ * -ERANGE is returned when the string is out of range for int value
+ * -EINVAL is returned when the string isn't a valid number or has any invalid
+ * trailing character.
+*/
+int strtoi_err(const char *str, int *value)
+{
+	char *endptr;
+	int err = strtoi_err_endptr(str, &endptr, value);
+
+	if (err) {
+		return err;
+	}
+
+	if (*endptr != '\0' && !isspace(*endptr)) {
+		return -ERANGE;
```

Comment:
Is this supposed to be -EINVAL?

Reply:
Good point. It makes more sense to use EINVAL. I'll change that